src/ImageSharp/Formats/Tiff/PhotometricInterpretation/BlackIsZero16TiffColor{TPixel}.cs

```diff
@@ -32,15 +32,25 @@ public override void Decode(ReadOnlySpan<byte> data, Buffer2D<TPixel> pixels, in
             var l16 = default(L16);
             for (int y = top; y < top + height; y++)
             {
-                for (int x = left; x < left + width; x++)
+                if (this.isBigEndian)
                 {
-                    ushort intensity = TiffUtils.ConvertToShort(data.Slice(offset, 2), this.isBigEndian);
-                    offset += 2;
+                    for (int x = left; x < left + width; x++)
+                    {
+                        ushort intensity = TiffUtils.ConvertToShortBigEndian(data.Slice(offset, 2));
+                        offset += 2;
 
-                    l16.PackedValue = intensity;
-                    color.FromL16(l16);
+                        pixels[x, y] = TiffUtils.ColorFromL16(l16, intensity, color);
```

Comment:
Are we using the `[x, y]` indexer like this elsewhere? We should process per row span to avoid the additional per-pixel multiplication.

Reply:
yeah your right. There was other places used the indexer which i have changed, too. See 0ae95f17

Reply:
weird, now a test fails only on `netcore2.1`: TiffDecoder_CanDecode_48Bit

Reply:
it only fails for `netcore2.1` and only in Release mode, not in Debug mode.
output looks like this:
![output](https://user-images.githubusercontent.com/38701097/128020409-32700238-5e95-400d-a20f-49d69cd70e6b.png)

Reply:
its again the bug with the defaults i reported recently: [netcore2.1 default bug](https://github.com/dotnet/runtime/issues/55623) 

I will try to work around it, but there are several places were defaults are used in the TiffColorDecoders

Reply:
I have changed the default usages introduced with this PR with values defined in `TiffUtils` now to workaround the issue with netcore2.1

Reply:
Man, that bug is annoying!